src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -243,12 +243,13 @@ protected override CompletionItemRules GetCompletionItemRules(ImmutableArray<(IS
             if (symbol.IsKind(SymbolKind.Method))
             {
                 var isInferredTypeDelegate = context.InferredTypes.Any(type => type.IsDelegateType());
-                if (!isInferredTypeDelegate)
+                if (!context.IsNameOfContext && !isInferredTypeDelegate)
```

Comment:
💭 Makes me wonder if the function pointer syntax `&MethodName` would be a problem

Reply:
This might be another issue, for this case:
```using System;
public unsafe class AA
{
    private static void CC()
    {
        delegate*<void> p = &$$
    }

    public static void Bar() {}               
}

here SyntaxContext.InferredType is empty.

Reply:
Create a bug here https://github.com/dotnet/roslyn/issues/50764